docs/modules/getting-started/pages/push.adoc

```diff
@@ -0,0 +1,112 @@
+= AeroGear Service Push SDK
+
+AeroGear for Android provides support for integrating Firebase Cloud Messaging (FCM) with the AeroGear Unified Push Server.
+
+Push messaging is a mechanism for sending data to mobile clients from a remote server. For mobile applications, this reduces battery usage, removes polling code and provides for a better user experience as data changes are communicated in real time.
+
+Messages that FCM receives from an application server are sent to the Android device over a connection managed by the operating system. These messages are then extracted and passed to an application using services definied provided by the AeroGear library.
+
+== Adding dependency
+
+Add the push dependency to your application module
+
+----
+implementation "org.aerogear:aerogear-push:[version]"
+----
+
+== Configuration file
+
+A `mobile-services.json` file must exist in the apps assets directory. It should specify configuration
+for link:https://github.com/aerogear/aerogear-unifiedpush-server/[Unified Push Server]. This configuration can be generated by the link:https://github.com/aerogear/mobile-cli[AeroGear Mobile CLI].
+
+For an example of push configuration see link:../../example/src/main/assets/mobile-services.json[example apps mobile-services.json].
+
+== Registering with Unified Push Server
+
+This example demonstrates how to register a device to provide the values required to register the application with FCM and Unified Push and perform the registration.
+
+[source,java]
+----
+PushService pushService = MobileCore.getInstance().getService(PushService.class);
```

Comment:
There is not the best place to add that since it's a singleton... you can do it when wherever you want how many times you want ;)

Reply:
I know but noob doesn't know that and first thing he is thiking. Where I should put this stuff. Application or Activity onCreate is the best place probably.

Reply:
There are several places registration can happen.  If you want to always be registered then you add it in the Application class.  If you want users to have control over it then you need to add it into a Activity.  There are reasons for both and those reasons are different and nuanced.

I think that the best place to introduce people to it is in an event handle on the activity (ie handleRegisterButtonPress or something) because we want to get peoeple used to the idea that push messages are opt in.

TL;DR; wrap it in an activity in the code sample and have register and unregister be in methods that are event handlers, thoughts?

Reply:
I got you both! My point is... IMO we should not mention in the doc _where_  developer should do that and let he/she free to do it wherever he/she wants

@secondsun The activity example you mention is exactly what we have in the example app. The developer can always consult it ;)

Reply:
@danielpassos Sounds good.